### PR TITLE
TVIST1-588: Agenda meeting point changes

### PR DESCRIPTION
**Pull request by jekuaitk**
* Added meeting point property to Agenda
* Added checks for agenda having date, start time, end time and meeting point before allowing status ready (Klar).
* Automatically fills out address meeting point on `AgendaCaseItem` if it is to be inspected.

https://jira.itkdev.dk/browse/TVIST1-588

### CODE REVIEW

File: assets/agenda_item/new.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* global $ */
 
 const $type = $('#agenda_item_type')
-$type.change(function () {
+$type.on('change', function () {
```
Comment (rimi-itk) [inline assets/agenda_item/new.js:4]: 👍

File: src/Controller/AgendaController.php [hunk 2]
```diff
@@ -252,7 +252,38 @@ public function edit(Agenda $agenda, Request $request): ?Response
...
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a end time as ready', [], 'agenda'));
+                }
+
+                if (empty($agenda->getAgendaMeetingPoint())) {
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a meeting point as ready', [], 'agenda'));
+                }
+
+                // Check if any flashes has been added
+                $endingFlashes = sizeof($request->getSession()->getFlashBag()->keys());
+                if ($endingFlashes > $startingFlashes) {
```
Comment (rimi-itk) [inline src/Controller/AgendaController.php:280]: You should not use (the number of) flash messages for control flow – you assume something about how the flash messages are handled internally and you cannot rely on that.

Use a boolean variable, `hasMessages`, say, or collect the messages in a list and check the length of this list, e.g.

```php
$messages = [];

if (empty(…)) {
  $messages['warning'] = new TranslatableMessage(…);
}

if (!empty($messages)) {
  foreach ($messages as $type => $msgs) {
    $this->addFlash($type, $message);
  }
  return $this->redirectToRoute(…);
}
```

File: src/Controller/CaseController.php [hunk 3]
```diff
@@ -779,4 +779,24 @@ public function applyIdentifierData(IdentificationHelper $identificationHelper,
...
+    /**
+     * @Route("/get/inspection-address", name="case_get_inspection_address", methods={"GET", "POST"})
+     */
+    public function getInspectionAddress(CaseEntityRepository $repository, AddressHelper $addressHelper, Request $request): Response
+    {
+        $identifier = $request->request->get('identifier');
+
+        if (empty($identifier)) {
+            return new JsonResponse(['address' => '']);
+        }
```
Comment (rimi-itk) [inline src/Controller/CaseController.php:792]: I'm not sure this micro optimization is worthwhile. I would just pass (the possible empty) identifier to `findOneBy` and let the check for no case return the empty address.

File: src/Entity/Agenda.php [hunk 4]
```diff
@@ -86,6 +86,12 @@ class Agenda implements LoggableEntityInterface
      */
     private $agendaBroadcasts;
 
+    /**
+     * @ORM\Column(type="string", length=255, nullable=true)
```
Comment (rimi-itk) [inline src/Entity/Agenda.php:90]: Are 255 characters enough?

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Some minor changes needed.

File: src/Entity/Agenda.php (hunk 4 above)
Comment (jekuaitk) [reply src/Entity/Agenda.php:90]: I would of thought so, however let's change it to type `text`.

File: src/Controller/CaseController.php (hunk 3 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:792]: Sure, removed.

File: src/Controller/AgendaController.php (hunk 2 above)
Comment (jekuaitk) [reply src/Controller/AgendaController.php:280]: Updated as proposed.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**